src/core/Akka/Dispatch/MessageQueues/UnboundedPriorityMessageQueue.cs

```diff
@@ -14,9 +15,11 @@ namespace Akka.Dispatch.MessageQueues
     /// <summary> 
     /// Base class for a message queue that uses a priority generator for messages 
     /// </summary>
-    public class UnboundedPriorityMessageQueue : BlockingMessageQueue
+    public class UnboundedPriorityMessageQueue : BlockingMessageQueue, IUnboundedDequeBasedMessageQueueSemantics
```

Comment:
Welp, I think my final thoughts on the subject are: let the users worry about their message orders. If they want to use stashing and priority mailboxes where the results aren't 100% predictable or stable, that's up to them.